src/Swift.Bindings/tests/CryptoKit/Foundation.Source.cs

```diff
@@ -0,0 +1,274 @@
+    }
+
+    /// <summary>
+    /// Swift runtime helper methods in C#.
+    /// </summary>
+    public static class Runtime
+    {
+        /// <summary>
+        /// https://github.com/apple/swift/blob/main/include/swift/ABI/MetadataValues.h#L117
+        /// </summary>
+        [Flags]
+        public enum ValueWitnessFlags
+        {
+            AlignmentMask = 0x0000FFFF,
+            IsNonPOD = 0x00010000,
+            IsNonInline = 0x00020000,
+            HasSpareBits = 0x00080000,
+            IsNonBitwiseTakable = 0x00100000,
+            HasEnumWitnesses = 0x00200000,
+            Incomplete = 0x00400000,
+        }
+
+        /// <summary>
+        /// See https://github.com/apple/swift/blob/main/include/swift/ABI/ValueWitness.def
+        /// </summary>
+        [StructLayout (LayoutKind.Sequential)]
+        public ref struct ValueWitnessTable
+        {
+            public IntPtr InitializeBufferWithCopyOfBuffer;
+            public IntPtr Destroy;
+            public IntPtr InitWithCopy;
+            public IntPtr AssignWithCopy;
+            public IntPtr InitWithTake;
+            public IntPtr AssignWithTake;
+            public IntPtr GetEnumTagSinglePayload;
+            public IntPtr StoreEnumTagSinglePayload;
+            private IntPtr _Size;
+            private IntPtr _Stride;
+            public ValueWitnessFlags Flags;
+            public uint ExtraInhabitantCount;
+            public int Size => _Size.ToInt32();
+            public int Stride => _Stride.ToInt32();
+            public int Alignment => (int)((Flags & ValueWitnessFlags.AlignmentMask) + 1);
+            public bool IsNonPOD => Flags.HasFlag (ValueWitnessFlags.IsNonPOD);
+            public bool IsNonBitwiseTakable => Flags.HasFlag (ValueWitnessFlags.IsNonBitwiseTakable);
+            public bool HasExtraInhabitants => ExtraInhabitantCount != 0;
+        }
+
+        public static unsafe void* GetMetadata<T>(T type) where T: ISwiftObject
+        {
+            return T.Metadata;
+        }
+
+        public static unsafe void* GetValueWitnessTable(void* metadata)
+        {
+            void* valueWitnessTable = (void*)Marshal.ReadIntPtr((IntPtr)metadata, -IntPtr.Size);
+            return valueWitnessTable;
+        }
+
+        public static unsafe void* GetConformanceDescriptor(string symbol)
+        {
+            IntPtr handle = IntPtr.Zero;
+            try
+            {
+                handle = NativeLibrary.Load(Foundation.Path);
+                void* conformanceDescriptor = NativeLibrary.GetExport(handle, symbol).ToPointer();
+                return conformanceDescriptor;
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException($"Failed to get conformance descriptor for symbol: {symbol}", ex);
+            }
+            finally
+            {
+                if (handle != IntPtr.Zero)
+                {
+                    NativeLibrary.Free(handle);
```

Comment:
Two things:
1 - will this free an already loaded library?
2 - you don't need an IntPtr.Zero check, this is documented as doing nothing on IntPtr.Zero

Reply:
The intention is to free the loaded library only if `NativeLibrary.GetExport` fails, which is why `handle != IntPtr.Zero` is used.